tests/integration/tools/mongodb/read/count.test.ts

```diff
@@ -0,0 +1,118 @@
+                required: false,
+            },
+            ...dbOperationParameters,
+        ]);
+    });
+
+    describe("with invalid arguments", () => {
+        const args = [
+            {},
+            { database: 123, collection: "bar" },
+            { foo: "bar", database: "test", collection: "bar" },
+            { collection: [], database: "test" },
+            { collection: "bar", database: "test", query: "{ $gt: { foo: 5 } }" },
+        ];
+        for (const arg of args) {
+            it(`throws a schema error for: ${JSON.stringify(arg)}`, async () => {
+                await connect(client(), cluster());
+                try {
+                    await client().callTool({ name: "count", arguments: arg });
+                    expect.fail("Expected an error to be thrown");
+                } catch (error) {
+                    expect(error).toBeInstanceOf(McpError);
+                    const mcpError = error as McpError;
+                    expect(mcpError.code).toEqual(-32602);
+                    expect(mcpError.message).toContain("Invalid arguments for tool count");
+                }
+            });
+        }
+    });
+
+    it("returns 0 when database doesn't exist", async () => {
+        await connect(client(), cluster());
+        const response = await client().callTool({
+            name: "count",
+            arguments: { database: "non-existent", collection: "foos" },
+        });
+        const content = getResponseContent(response.content);
+        expect(content).toEqual('Found 0 documents in the collection "foos"');
+    });
+
+    it("returns 0 when collection doesn't exist", async () => {
+        await connect(client(), cluster());
+        const mongoClient = cluster().getClient();
+        await mongoClient.db(randomDbName).collection("bar").insertOne({});
+        const response = await client().callTool({
+            name: "count",
+            arguments: { database: randomDbName, collection: "non-existent" },
+        });
+        const content = getResponseContent(response.content);
+        expect(content).toEqual('Found 0 documents in the collection "non-existent"');
+    });
+
+    describe("with existing database", () => {
+        beforeEach(async () => {
+            const mongoClient = cluster().getClient();
+            await mongoClient
+                .db(randomDbName)
+                .collection("foo")
+                .insertMany([
+                    { name: "Peter", age: 5 },
+                    { name: "Parker", age: 10 },
+                    { name: "George", age: 15 },
+                ]);
+        });
+
+        const testCases = [
+            { filter: undefined, expectedCount: 3 },
+            { filter: {}, expectedCount: 3 },
+            { filter: { age: { $lt: 15 } }, expectedCount: 2 },
+            { filter: { age: { $gt: 5 }, name: { $regex: "^P" } }, expectedCount: 1 },
+        ];
+        for (const testCase of testCases) {
```

Comment:
```suggestion
        for (const testCase of [
            { filter: undefined, expectedCount: 3 },
            { filter: {}, expectedCount: 3 },
            { filter: { age: { $lt: 15 } }, expectedCount: 2 },
            { filter: { age: { $gt: 5 }, name: { $regex: "^P" } }, expectedCount: 1 },
        ]) {
```
nit: I think this is quicker to read

Reply:
I initially had it like that, but then moved it out because I thought it's getting cluttered with the test case definition inlined into the for loop.